modeshape-jcr/src/main/java/org/modeshape/jcr/query/engine/ScanningQueryEngine.java

```diff
@@ -1537,6 +1538,9 @@ protected RowFilter createFilter() {
                     return new DynamicOperandFilter(operation) {
                         @Override
                         protected boolean evaluate( Object leftHandValue ) {
+                            if (Path.class.isAssignableFrom(defaultType.getType())) {
+                                leftHandValue = pathFactory.create(leftHandValue);
```

Comment:
Yes, but that's not the method this code is calling. Since `leftHandValue` is of type `Object`, then `ValueFactory.create(Object)` will be the method that gets called, and the implementation doesn't currently check for array.